holoplot/__init__.py

```diff
@@ -0,0 +1,385 @@
+        except ImportError:
+            raise ImportError('Could not patch plotting API onto streamz. '
+                              'Streamz could not be imported.')
+        sdf.DataFrame.plot = property(_patch_plot)
+        sdf.DataFrames.plot = property(_patch_plot)
+        sdf.Series.plot = property(_patch_plot)
+        sdf.Seriess.plot = property(_patch_plot)
+    if 'pandas' in library:
+        try:
+            import pandas as pd
+        except:
+            raise ImportError('Could not patch plotting API onto pandas. '
+                              'Pandas could not be imported.')
+        pd.DataFrame.plot = property(_patch_plot)
+        pd.Series.plot = property(_patch_plot)
+    if 'dask' in library:
+        try:
+            import dask.dataframe as dd
+        except:
+            raise ImportError('Could not patch plotting API onto dask. '
+                              'Dask could not be imported.')
+        dd.DataFrame.plot = property(_patch_plot)
+        dd.Series.plot = property(_patch_plot)
+    if 'intake' in library:
+        try:
+            from intake.source.base import DataSource
+        except ImportError:
+            raise ImportError('Could not patch plotting API onto intake. '
+                              'Intake could not be imported.')
+        DataSource.plot = property(_patch_plot)
+    if 'xarray' in library:
+        try:
+            from xarray import DataArray
+        except ImportError:
+            raise ImportError('Could not patch plotting API onto xarray. '
+                              'Xarray could not be imported.')
+        DataArray.plot = property(_patch_plot)
+    if extension:
+        _extension(extension)
+
+
+class HoloPlot(object):
+
+    def __init__(self, data):
+        self._data = data
+
+    def __call__(self, x=None, y=None, kind=None, backlog=1000,
+             width=700, height=300, title=None, grid=False,
+             legend=True, logx=False, logy=False, loglog=False,
+             xticks=None, yticks=None, xlim=None, ylim=None, rot=None,
+             fontsize=None, colormap=None, hover=True, **kwds):
+        converter = HoloViewsConverter(
+            self._data, x, y, width=width, height=height, backlog=backlog,
+            title=title, grid=grid, legend=legend, logx=logx,
+            logy=logy, loglog=loglog, xticks=xticks, yticks=yticks,
+            xlim=xlim, ylim=ylim, rot=rot, fontsize=fontsize,
+            colormap=colormap, hover=hover, kind=kind, **kwds
+        )
+        return converter(kind, x, y)
+
+    def line(self, x=None, y=None, **kwds):
+        """
+        Line plot
+
+        Parameters
+        ----------
+        x, y : label or position, optional
+            Coordinates for each point.
+        **kwds : optional
+            Keyword arguments to pass on to
+            :py:meth:`intake.source.base.DataSource.plot`.
+        Returns
+        -------
+        Element : Element or NdOverlay of Elements
```

Comment:
Given that holoplot is for returning an HV object from something that is not an HV object, should it convey that the underlying library is switching by saying `hv.Element`, etc. here?  Otherwise people may have no idea where to look for Element.

Reply:
I'll have to update these, they aren't really accurate anymore, it could be any one of: Element/NdOverlay/HoloMap/DynamicMap/NdLayout/GridSpace

Reply:
Maybe just "Returns _ HoloViews Object (any one of ...)", then.